10up/ElasticPress
# Feature/post type enhancements

## Description of the Change

This PR tries to address two problems, related to `post_type` support. 

* In order to restrict the autosuggest query by post_type. I have added a control that checks `[name="post_type"]` input and applying particular `post_type` to query. Therefore, we can show only the "products" for WooCommerce's own search bar. 

* I have changed facet's query build logic a bit, now `ep_facet_allowed_query_args` allows to extend allowed query parameters. `post_type` and `s` are supported by default. 

### Alternate Designs

* Perhaps, we can add a new flag item for Autosuggest's localize script to understand that request made from a product page

### Benefits

Improving `post_type` support without adding custom code

### Possible Drawbacks

* I don't consider as drawbacks but we likely need to re-apply autosuggest related changes to #1663

### Verification Process

[Autosuggest]
* Install WooCommerce
* Activate EP's Woo and Autosuggest feature
* Visit `/shop/` page
* Enter the search query and see the suggested results 

[Facets]
* Make a WooCommerce product search
* Check facet's widgets have `post_type` on the target url

### Screenshot
(first input Woo's search, second is generic) 

<img width="412" alt="Screen Shot 2020-03-05 at 15 49 25" src="https://user-images.githubusercontent.com/1421387/75984851-36a87280-5efc-11ea-86df-9863e23690f9.png">


### Checklist:



### Applicable Issues

#1687, #1684

### Changelog Entry

* Autosuggest: respects the `[name="post_type"]` input in the same form.
* Additional query parameters can register through `ep_facet_allowed_query_args` for facets
